addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -602,3 +345,194 @@ def action_see_accrual_plans(self):
+            date = datetime.fromisoformat(date).date()
+        elif date and isinstance(date, datetime):
+            date = date.date()
+        elif not date:
+            date = fields.Date.today()
+
+        allocations_leaves_consumed = self.get_leaves_consumed(employees)
+
+        for employee in employees:
+            for leave_type in self.filtered(lambda lt: lt.requires_allocation == 'yes'):
+                if len(allocations_leaves_consumed[employee][leave_type].items()) == 0:
+                    continue
+                lt_info = (
+                    leave_type.name,
+                    {
+                        'remaining_leaves': 0,
+                        'virtual_remaining_leaves': 0,
+                        'max_leaves': 0,
+                        'accrual_bonus': 0,
+                        'leaves_taken': 0,
+                        'virtual_leaves_taken': 0,
+                        'leaves_requested': 0,
+                        'leaves_approved': 0,
+                        'closest_allocation_remaining': 0,
+                        'closest_allocation_expire': False,
+                        'request_unit': leave_type.request_unit,
+                        'icon': leave_type.sudo().icon_id.url,
+                    },
+                    leave_type.requires_allocation,
+                    leave_type.id)
+                leave_allocations = self.env['hr.leave.allocation']
+                for allocation, data in allocations_leaves_consumed[employee][leave_type].items():
+                    future_leaves = 0
+                    if allocation.allocation_type == 'accrual':
+                        future_leaves = allocation.get_future_leaves_on(date)
+                    if not allocation or (not allocation.date_to or allocation.date_to >= date) and allocation.date_from <= date:
+                        lt_info[1]['remaining_leaves'] += data['remaining_leaves'] + future_leaves
+                        lt_info[1]['virtual_remaining_leaves'] += data['virtual_remaining_leaves'] + future_leaves
+                        lt_info[1]['max_leaves'] += data['max_leaves'] + future_leaves
+                        lt_info[1]['accrual_bonus'] += future_leaves
+                        lt_info[1]['leaves_taken'] += data['leaves_taken']
+                        lt_info[1]['virtual_leaves_taken'] += data['virtual_leaves_taken']
+                        lt_info[1]['leaves_requested'] += data['virtual_leaves_taken']
+                        lt_info[1]['leaves_approved'] += data['leaves_taken']
+                        leave_allocations |= allocation
+                sorted_leave_allocations = leave_allocations.filtered('date_to').sorted(key='date_to')\
+                    + leave_allocations.filtered(lambda a: not a.date_to)\
+                    if leave_allocations else False
+                closest_allocation = sorted_leave_allocations[0] if sorted_leave_allocations else self.env['hr.leave.allocation']
```

Comment:
I could be wrong, but on one glance :innocent: 

closest_allocation_expire is the allocation that is the closest one and still has remaining leaves. This is because we don't want to see in the dashboard something like - 20 days.. 0 valid until xxxxxxx.... in case we have two allocations, and one with 0 days remaining. I don't see if we filter it out here. In previous code in was done here -https://github.com/odoo/odoo/pull/108148/files#diff-3da99c16f7aebde2f17805efa4363d02d634cc4410f43c90d41eec9e88b0dcb4L325